### PR TITLE
Use ActorSystem's ThreadFactory for the Aeron threads, #27603

### PR DESCRIPTION
**Pull request by patriknw**
Refs #27603

### CODE REVIEW

File: akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala [hunk 1]
```diff
@@ -24,14 +24,13 @@ class DaemonicSpec extends AkkaSpec {
...
-        ConfigFactory.parseString(
-          """
+        ConfigFactory.parseString("""
         akka.daemonic = on
         akka.actor.provider = remote
-        akka.remote.classic.netty.tcp.transport-class = "akka.remote.transport.netty.NettyTransport"
         akka.remote.classic.netty.tcp.port = 0
         akka.remote.artery.canonical.port = 0
         akka.log-dead-letters-during-shutdown = off
+        #akka.remote.artery.advanced.aeron.idle-cpu = 5
```
Comment (patriknw) [inline akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala]: I confirmed that test is passing with all 1-10 values

**Review by johanandren — APPROVED** [review]
LGTM,

Before it would spawn threads with its own factory?

**Comment by patriknw** [comment]
> Before it would spawn threads with its own factory?

yes, and nothing fancy, just `Thread::new`

**Review by helena — APPROVED** [review]
LGTM

**Pull request merged**